doc/continuous-integration.md

```diff
@@ -0,0 +1,150 @@
+CI/CD pipelines and jobs. 
+
+Note (for geeks): If you are interested in Zuul CI platform and want to deploy your own development instance of it,
+then read the official [quick-start](https://zuul-ci.org/docs/zuul/latest/tutorials/quick-start.html) manual
+or visit [this](https://github.com/matofederorg/zuul-config) tutorial which aims a connection
+of Zuul CI platform with a GitHub organization.
+
+## Configuration
+
+SCS Zuul automatically recognizes `.zuul.yaml` configuration file that is located in the
+k8s-cluster-api-provider's root. This file informs Zuul about the project's [default-branch](https://zuul-ci.org/docs/zuul/latest/config/project.html#attr-project.default-branch) and 
+preferred [merge-mode](https://zuul-ci.org/docs/zuul/latest/config/project.html#attr-project.merge-mode).
+It also references [SCS Zuul pipelines](https://github.com/matofederorg/zuul-config) and
+their jobs used by the k8s-cluster-api-provider project. Then, jobs link Ansible playbooks that contain
+tasks for actual CI testing. 
+
+See relevant CI configuration files:
+```text
+├── .zuul.yaml
+├── playbooks
+│   ├── cleanup.yaml
+│   ├── dependencies.yaml
+│   ├── e2e.yaml
+│   ├── templates
+│   │   └── environment.tfvars.j2
+```
+
+## Pipelines
+
+This section describes an [SCS Zuul pipelines](https://github.com/SovereignCloudStack/zuul-config/blob/main/zuul.d/gh_pipelines.yaml) that are used by the k8s-cluster-api-provider project.
+
+- `e2e-test`
+  - It is triggered by the `e2e-test` label in the opened PR
+  - It executes `e2e-conformance` job
+  - It applies the PR label `successful-e2e-test` and leaves an informative PR comment when the `e2e-conformance` job succeeded
+  - It applies the PR label `failed-e2e-test` and leaves an informative PR comment when the `e2e-conformance` job failed
+  - It applies the PR label `cancelled-e2e-test` and leaves an informative PR comment when the `e2e-conformance` job is canceled
+
+- `unlabel-on-update-e2e-test`
+  - It is triggered by the PR update only when PR contains the `successful-e2e-test` label
+  - It ensures that any PR update invalidates a previous successful e2e test
+  - It removes `successful-e2e-test` label from the PR
+
+- `e2e-quick-test`
+  - It is triggered by the `e2e-quick-test` label in the opened PR
+  - It executes `e2e-quick` job
+  - It applies the PR label `successful-e2e-quick-test` and leaves an informative PR comment when the `e2e-quick` job succeeded
+  - It applies the PR label `failed-e2e-quick-test` and leaves an informative PR comment when the `e2e-quick` job failed
+  - It applies the PR label `cancelled-e2e-quick-test` and leaves an informative PR comment when the `e2e-quick` job is canceled
+
+- `unlabel-on-update-e2e-quick-test`
+  - It is triggered by the PR update only when PR contains the `successful-e2e-quick-test` label
+  - It ensures that any PR update invalidates a previous successful e2e test
+  - It removes `successful-e2e-quick-test` label from the PR
+
```

Comment:
I think it should be mentioned that the unlabel jobs are defined in the Zuul config repo?

Reply:
Find the sentence at the top of the `Pipelines` section:

`This section describes an [SCS Zuul pipelines](https://github.com/SovereignCloudStack/zuul-config/blob/main/zuul.d/gh_pipelines.yaml) that are used by the k8s-cluster-api-provider project.`

It already links the Zuul config repository as a source for all mentioned pipelines.